### PR TITLE
docs: add tools page

### PR DESCRIPTION
**Pull request by richtia**
Add a new page linking to existing tools that support substrait.

### CODE REVIEW

File: site/docs/tools/_config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+arrange:
+  - producer_tools.md
+  - substrait_validator.md
```
Comment (westonpace) [inline site/docs/tools/_config:3]: Missing third_party_tools.md?

File: site/docs/tools/producer_tools.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Producer Tools
```
Comment (westonpace) [inline site/docs/tools/producer_tools.md:1]: Are these linked into the main site automatically somehow?

File: site/docs/tools/_config (hunk 1 above)
Comment (richtia) [reply site/docs/tools/_config:3]: Added

File: site/docs/tools/producer_tools.md (hunk 2 above)
Comment (richtia) [reply site/docs/tools/producer_tools.md:1]: Yes, mkdocs should handle everything.

**Comment by westonpace** [comment]
Can you rebase/merge (I'm pretty sure I can do it but don't want to mess up your branch if you're working on it):

image

**Comment by richtia** [comment]
Done!

File: site/docs/tools/producer_tools.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+## Isthmus
+
+[Substrait Isthmus](https://github.com/substrait-io/substrait-java/tree/main/isthmus) is a Java 
+library which enables serializing SQL to [Substrait Protobuf](https://substrait.io/serialization/binary_serialization/) 
+via the Calcite SQL compiler.
```
Comment (westonpace) [inline site/docs/tools/producer_tools.md:7]:
```suggestion
[Isthmus](https://github.com/substrait-io/substrait-java/tree/main/isthmus) is an application 
that serializes SQL to [Substrait Protobuf](https://substrait.io/serialization/binary_serialization/) 
via the Calcite SQL compiler.
```
I think the fact that it uses Java under the hood is irrelevant.  It tries (via graal) to produce a simple executable so the user would never have to know that they were using Java.  Now, it's not easily built and requires cloning substrait-java but I think those details are irrelevant in a tool description.

File: site/docs/tools/third_party_tools.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Third Party Tools
+
+## Substrait-tools
+The [substrait-tools](https://pypi.org/project/substrait-tools/) python package provides 
+a command line interface for producing/consuming substrait plans by leveraging the APIs
+from different producers and consumers. Source code can be found [here](https://github.com/richtia/substrait-tools).
```
Comment (westonpace) [inline site/docs/tools/third_party_tools.md:6]:
```suggestion
from different producers and consumers.
```

Let's leave the "source code can be found here" for pypi (as an aside, you can configure pypi to show a github link by using project links https://stackoverflow.com/questions/61156290/how-to-set-project-links-in-pypi).

**Review by westonpace — CHANGES_REQUESTED** [review]
Ok, I built locally.  The pages seem a little sparse but that is probably fine for now.  Maybe in the future we can add a few usage examples for each one?

File: site/docs/tools/third_party_tools.md (hunk 4 above)
Comment (richtia) [reply site/docs/tools/third_party_tools.md]: Cool, just configured it.

**Review by westonpace — APPROVED** [review]

**Pull request merged**